Cargo.toml

```
@@ -48,13 +48,17 @@ log = "0.4.8"
 [target.'cfg(unix)'.dependencies]
 libc = "0.2.121"
 
+[target.'cfg(windows)'.dependencies]
+rand = "0.8"
```

It seems like this is used only in tests?
```suggestion
[target.'cfg(windows)'.dev-dependencies]
rand = "0.8"
```

---

It's used to emulate `socketpair` functionality on Windows because there's unfortunately no equivalent system call. The best we can do is create a temporary file, connect through it, and then delete it. I create the file by generating a random temp file name (https://github.com/sullivan-sean/mio/blob/488254d2b509b4b1152ee118aabf8533616b5d47/src/sys/windows/uds/stdnet/net.rs#L24-L28)

---

I'd personally recommend using `getrandom` instead of importing the entire `rand` crate.

---

Since it looks like `getrandom` just abstracts across platforms and we only need this in windows we could also just directly use `BCryptGenRandom` from `windows_sys`? (this is the same approach as [in `getrandom`](https://github.com/rust-random/getrandom/blob/0b71d50b4975b492c7ed7185d5d366403966c01b/src/windows.rs#L29))